app/src/org/commcare/android/view/GraphView.java

```diff
@@ -127,6 +128,24 @@ private void setPanAndZoom(boolean allow) {
         mRenderer.setPanEnabled(allow, allow);
         mRenderer.setZoomEnabled(allow, allow);
         mRenderer.setZoomButtonsVisible(allow);
+        if (allow) {
+            DefaultRenderer.Location loc = stringToLocation(mData.getConfiguration("zoom-location", "bottom-right"));
+            mRenderer.setZoomLocation(loc);
+        }
+    }
+
+    private DefaultRenderer.Location stringToLocation(String str) {
+        // Java, why don't you have a concise way to define a hashmap?

```

Comment:
As of Java 7 you can match strings using switch statements. Doing so would clean things up a little bit.